File: code-samples/voice/call-log.php
<?php
// Remember to modify the path ./../ pointing to the location where the RingCentral SDK was installed and the .env file was saved!
require('./../vendor/autoload.php');
$dotenv = Dotenv\Dotenv::createImmutable(__DIR__ . './../');
$dotenv->load();

# Instantiate the SDK and get the platform instance
$rcsdk = new RingCentral\SDK\SDK( $_ENV['RC_CLIENT_ID'],
                                  $_ENV['RC_CLIENT_SECRET'],
                                  $_ENV['RC_SERVER_URL'] );
$platform = $rcsdk->platform();

// Authenticate a user using a personal JWT token
try {
  $platform->login( [ "jwt" => $_ENV['RC_JWT'] ] );
  read_user_calllog();
} catch (\RingCentral\SDK\Http\ApiException $e) {
  exit("Unable to authenticate to platform. Check credentials. " . $e->message . PHP_EOL);
}

/*
*  Read user call log between a period of time
*/
function read_user_calllog() {
  global $platform;
  try{
    $queryParams = array(
      'dateFrom' => "2024-01-01T00:00:00.000Z",
      'dateTo' => "2024-01-31T23:59:59.009Z",
      'view' => "Detailed"
    );

    $endpoint = "/restapi/v1.0/account/~/extension/~/call-log";
    $resp = $platform->get($endpoint, $queryParams);
    foreach ($resp->json()->records as $record) {
      print_r (json_encode($record, JSON_PRETTY_PRINT));
    }
  }catch (\RingCentral\SDK\Http\ApiException $e) {
    print_r ("Unable to read user call log. " . $e->getMessage() . PHP_EOL);
  }
}
?>

File: code-samples/voice/call-log.rb
require 'ringcentral'
require 'dotenv/load'

# Remember to modify the path to where you saved your .env file!
Dotenv.load("./../.env")

#
# Read user call log between a period of time
#
def read_user_calllog()
  begin
    endpoint = "/restapi/v1.0/account/~/extension/~/call-log"
    queryParams = {
      'dateFrom': "2024-01-01T00:00:00.000Z",
      'dateTo': "2024-01-31T23:59:59.009Z",
      'view': "Detailed"
    }
    resp = $platform.get(endpoint, queryParams)
    for record in resp.body['records'] do
        puts JSON.pretty_generate(record)
    end
  rescue StandardError => e
    puts ("Unable to read user call log. " + e.to_s)
  end
end

# Instantiate the SDK and get the platform instance
$platform = RingCentral.new( ENV['RC_CLIENT_ID'], ENV['RC_CLIENT_SECRET'], ENV['RC_SERVER_URL'] )

# Authenticate a user using a personal JWT token
def login()
  begin
    $platform.authorize(jwt: ENV['RC_JWT'])
    read_user_calllog()
  rescue StandardError => e
    puts ("Unable to authenticate to platform. Check credentials." + e.to_s)
  end
end

login()
##############################
# End of quick start section
##############################
